consumer-test/integrationTestController.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+function fail() {
+    local message="$1"
+    local returnCode=${2:-1}
+    echo "[ERROR] ${message}" >&2
+    exit "${returnCode}"
+}
+
+function notify() {
+
+    local state=${1}
+    local description=${2}
+    local hash=${3}
+
+    echo "[INFO] Notifying about state \"${state}\" for commit \"${hash}\"."
+
+    curl -X POST \
+        --fail \
+        --silent \
+        --output /dev/null \
+        --data "{\"state\": \"${state}\", \"target_url\": \"${TRAVIS_BUILD_WEB_URL}\", \"description\": \"${description}\", \"context\": \"integration-tests\"}" \
+        --user "${INTEGRATION_TEST_VOTING_USER}:${INTEGRATION_TEST_VOTING_TOKEN}" \
+    "https://api.github.com/repos/SAP/jenkins-library/statuses/${hash}" || fail "Cannot send notification. curl return code: $?"
+}
+
+function cleanup() {
+    [ -z "${notificationThreadPid}" ] || kill -PIPE "${notificationThreadPid}" &>/dev/null
+}
+
+trap cleanup EXIT
+
+#
+# In case the build is performed for a pull request TRAVIS_COMMIT is a merge
+# commit between the base branch and the PR branch HEAD. That commit is actually built.
+# But for notifying about a build status we need the commit which is currenty
+# the HEAD of the PR branch.
+#
+# In case the build is performed for a simple branch (not associated with a PR)
+# In this case there is no merge commit between any base branch and HEAD of a PR branch.
+# The commit which we need for notifying about a build status is in this case simply
+# TRAVIS_COMMIT itself.
+#
+COMMIT_HASH_FOR_STATUS_NOTIFICATIONS="${TRAVIS_PULL_REQUEST_SHA}"
+[ -z "${COMMIT_HASH_FOR_STATUS_NOTIFICATIONS}" ] && COMMIT_HASH_FOR_STATUS_NOTIFICATIONS="${TRAVIS_COMMIT}"
+
+notify "pending" "Integration tests in progress." "${COMMIT_HASH_FOR_STATUS_NOTIFICATIONS}"
```

Comment:
When our build is triggered we set the "pending" status for our status for the current commit.
It is important to know that we notify in the strong sense not for a pull request, but always for a commit. In case this commit it the HEAD of a pull request that pull request will receive that status. This means: when we push a commit - which is also the HEAD of a PR - into the `it/*` branch namespace the PR will finally get the notifiction. This work even if the PR is proposed from a fork.